tests/assets/test_digital_assets.py

```diff
@@ -0,0 +1,163 @@
+
+    # `data` needs to be a dictionary
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.data = 'a'
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    # TODO: Check where to test for the amount
+    """
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 'a'
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 2
+    tx['transaction']['asset'].update({'divisible': False})
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(AmountError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 0
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(AmountError):
+        b.validate_transaction(tx_signed)
+    """
+
+@pytest.mark.usefixtures('inputs')
+def test_validate_transfer_asset_id_mismatch(b, user_vk, user_sk):
+    from bigchaindb_common.exceptions import AssetIdMismatch
+    from bigchaindb.models import Transaction
+
+    tx_create = b.get_owned_ids(user_vk).pop()
+    tx_create = b.get_transaction(tx_create.txid)
+    tx_transfer = Transaction.transfer(tx_create.to_inputs(), [user_vk], tx_create.asset)
+    tx_transfer.asset.data_id = 'aaa'
+    tx_transfer_signed = tx_transfer.sign([user_sk])
+    with pytest.raises(AssetIdMismatch):
+        tx_transfer_signed.validate(b)
+
+
+@pytest.mark.usefixtures('inputs')
+def test_get_asset_id_create_transaction(b, user_vk):
+    from bigchaindb.models import Asset
+
+    tx_create = b.get_owned_ids(user_vk).pop()
+    tx_create = b.get_transaction(tx_create.txid)
+    asset_id = Asset.get_asset_id(tx_create)
+
+    assert asset_id == tx_create.asset.data_id
+
+
+@pytest.mark.usefixtures('inputs')
+def test_get_asset_id_transfer_transaction(b, user_vk, user_sk):
+    from bigchaindb.models import Transaction, Asset
+
+    tx_create = b.get_owned_ids(user_vk).pop()
+    tx_create = b.get_transaction(tx_create.txid)
+    # create a transfer transaction
+    tx_transfer = Transaction.transfer(tx_create.to_inputs(), [user_vk], tx_create.asset)
+    tx_transfer_signed = tx_transfer.sign([user_sk])
+    # create a block
+    block = b.create_block([tx_transfer_signed])
+    b.write_block(block, durability='hard')
+    # vote the block valid
+    vote = b.vote(block.id, b.get_last_voted_block().id, True)
+    b.write_vote(vote)
+    asset_id = Asset.get_asset_id(tx_transfer)
+
+    assert asset_id == tx_transfer.asset.data_id
+
+
+@pytest.mark.usefixtures('inputs')
```

Comment:
Same here. We could just create the transactions programmatically (without the db).